hackathon-for-hunger/Modules/User/Views/ProfileViewController.swift

```diff
@@ -10,10 +10,50 @@ import UIKit
 
 class ProfileViewController: UIViewController {
     
+
+    private let userService: UserService = UserService()
+
+ 
+    @IBOutlet weak var userNameTF: UITextField!
+    @IBOutlet weak var emailTF: UITextField!
+    
+    @IBAction func save(sender: AnyObject) {
+        print("Save pressed")
+        updateUser()
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         self.setupMenuBar()
         self.title = "Profile"
+        let currentUser = AuthService.sharedInstance.getCurrentUser()
+        userNameTF.text = currentUser?.name
+        emailTF.text = currentUser?.email
+
+
+    }
+    
+    func updateUser() {
+        
+        
+        var user = UserRegistration()
+        user.email = emailTF.text ?? "donor@hacksmiths.com"
+        user.phone = "3211231234"
+        user.name = userNameTF.text ?? "please enter your name"
+        user.password = "password"
+        user.role = .Donor
+        
+        
+        
+        userService.updateUser(user).then() {
```

Comment:
As discussed from the main refactor, we are now follwing the Model View Presenter design pattern. Which means you shold not be calling the service directly in a view controller. This is a presenters Job. Please look at another modules setup to see whats been done already